x-pack/plugins/ml/public/data_frame/common/pivot_group_by.ts

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Dictionary } from '../../../common/types/common';
+
+export enum PIVOT_SUPPORTED_GROUP_BY_AGGS {
+  DATE_HISTOGRAM = 'date_histogram',
+  HISTOGRAM = 'histogram',
+  TERMS = 'terms',
+}
+
+type FieldName = string;
+
+interface GroupByConfigBase {
+  field: FieldName;
+  formRowLabel: string;
+}
+
+export const histogramIntervalFormatRegex = /^[0-9]+((\.)?[0-9]+)?$/;
+export const dateHistogramIntervalFormatRegex = /^[0-9]+(ms|s|m|h|d|w|M|q|y)$/;
```

Comment:
Does this mean a value like `0d` is accepted?

Reply:
Good catch! I adapted both regexp and added some more tests in e851cdd.